pkg/api/v1/clients/kube/crd/crd.go

```diff
@@ -103,16 +103,15 @@ func (d Crd) KubeResource(resource resources.InputResource) (*v1.Resource, error
 		}
 
 		delete(data, "metadata")
-		delete(data, "status")
+		delete(data, "namespacedStatuses")
```

Comment:
k8s expects and may treat fields such as `metadata`, `spec`, and `status` in a special way. can we keep the field called `status` at the top level, but handle both namespaced and non-namespaced statuses under the covers while unmarshalling? e.g.

```
status:
  <status>
```

or

```
status:
  ns1:
    <status1>
  ns2:
    <status2>
```

rather than
```
namespacedStatuses:
  ns1:
    <status1>
  ns2:
    <status2>
```

Reply:
Spoke offline (and will add comment to code):

We are still keeping the status as a top level field. What this line is doing is removing the status (now namespaced_statuses) field from the spec, so that when we persist it, it does not live both on the spec and the status.